bgpd/bgp_bfd.c

```diff
@@ -586,6 +591,11 @@ DEFUN(neighbor_bfd_profile, neighbor_bfd_profile_cmd,
 	if (!peer)
 		return CMD_WARNING_CONFIG_FAILED;
 
+	if (!peer->bfd_config) {
+		vty_out(vty, "%% Enable bfd for this neighbor first\n");
```

Comment:
NAK. If the user issues `neighbor 2001:db8:2::2 bfd profile slowtx` as a command, that configuration must be saved (while having no effect) even if BFD is not enabled for the neighbor. If `peer->bfd_config` does not exist, it must be created at this point even if BFD is disabled. (That also means checking for `peer->bfd_config` cannot be used to check if BFD is enabled on a peer.)

Reply:
Hmm, in that case the solution for OSPF is wrong as well? https://github.com/FRRouting/frr/commit/f3fd7196eb

Reply:
Can you please suggest any different way how to check that the bfd is or isn't enabled? It was discussed here https://github.com/FRRouting/frr/pull/10570#issuecomment-1069081853 and in the comments above that 'neighbor bfd' command should be issued before 'neighbor bfd profile' command.